examples/ReadWriteSplittingSample/src/main/java/software/amazon/ReadWriteSplittingSample.java

```diff
@@ -0,0 +1,271 @@
+        }
+
+        if (i < MAX_SIMULTANEOUS) {
+          // Space out initial threads to distribute workload across time.
+          TimeUnit.MILLISECONDS.sleep(THREAD_DELAY_MS);
+        }
+      }
+
+      executorService.shutdown();
+      LOGGER.info("Awaiting termination...");
+      boolean successfullyTerminated = executorService.awaitTermination(EXECUTOR_TIMEOUT_MINS, TimeUnit.MINUTES);
+      if (!successfullyTerminated) {
+        LOGGER.warning(String.format(
+            "The executor service timed out after waiting %d minutes for termination. " +
+                "Consider increasing the EXECUTOR_TIMEOUT_MINS value.", EXECUTOR_TIMEOUT_MINS));
+      }
+
+      if (APPROACH_ID == 3) {
+        LOGGER.info("Closing internal connection pools...");
+        ConnectionProviderManager.releaseResources();
+      }
+
+      long duration = System.nanoTime() - start;
+      LOGGER.info(String.format("Test completed in %dms", TimeUnit.NANOSECONDS.toMillis(duration)));
+    } catch (InterruptedException e) {
+      LOGGER.severe("The main thread was interrupted.");
+      throw new RuntimeException(e);
+    } finally {
+      deleteTables();
+    }
+  }
+
+  private static void deleteTables() throws SQLException {
+    String dropTableSql = "drop table if exists rw_sample_?";
+    try (Connection conn = DriverManager.getConnection(WRITER_CLUSTER, noRwProps);
+         PreparedStatement stmt = conn.prepareStatement(dropTableSql)) {
+      for (int i = 0; i < NUM_THREADS; i++) {
+        stmt.setInt(1, i);
+        stmt.execute();
+      }
+    }
+  }
+
+  private static HikariConfig getHikariConfig(HostSpec hostSpec, Properties props) {
+    final HikariConfig config = new HikariConfig();
+    config.setMaximumPoolSize(10);
+    return config;
+  }
+
+  private static void executeWrites(Connection conn, int tableNum) throws SQLException {
+    long start = System.nanoTime();
+
+    String createSql = "create table rw_sample_?(some_num int not null)";
+    String insertSql = "insert into rw_sample_? values (1)";
+    try (PreparedStatement createStmt = conn.prepareStatement(createSql);
+         PreparedStatement insertStmt = conn.prepareStatement(insertSql)) {
+      createStmt.setInt(1, tableNum);
+      createStmt.execute();
+
+      insertStmt.setInt(1, tableNum);
+      for (int i = 0; i < NUM_WRITES; i++) {
+        insertStmt.execute();
+      }
+    }
+
+    long duration = System.nanoTime() - start;
+    long durationMs = TimeUnit.NANOSECONDS.toMillis(duration);
+    LOGGER.finest(String.format("Thread %d write duration: %dms", tableNum, durationMs));
+  }
+
+  private static void executeReads(Connection conn, int tableNum) throws SQLException {
+    long start = System.nanoTime();
+    String selectSQL = "select * from rw_sample_?";
+
+    try (PreparedStatement stmt = conn.prepareStatement(selectSQL)) {
+      for (int i = 0; i < NUM_READS; i++) {
+        stmt.setInt(1, tableNum);
+        stmt.execute();
+      }
+    }
```

Comment:
I didnt try it out but I think you can do the following so you only execute once:
```suggestion
    try (PreparedStatement stmt = conn.prepareStatement(selectSQL)) {
      for (int i = 0; i < NUM_READS; i++) {
        stmt.setInt(1, tableNum);
        stmt.addToBatch();
      }
      stmt.executeBatch();
    }
```

Reply:
What's the advantage to doing it this way? If its just for performance, I wonder if we should make the change or not. I was using this code to purposely overload the writer so that we could observe how r/w splitting improved performance for this scenario. Then again, users probably won't be overloading their writer when they execute this code, since it requires a pretty specific configuration of settings in this file + rds cluster configuration. What do you think, should we go ahead with the change?

Reply:
It is a more common practice to use batch execute because of performance improvements. I just think that since it is a sample code we would want to follow best practices in our code. If we use batchExecute do we no longer see a performance improvement?